### PR TITLE
Improve usability of MetricProvider

### LINKED ISSUE #167: Add `addMetricProcessor` function to MetricProvider
https://cloud-native.slack.com/archives/C01NCHR19SB/p1618582139006900?thread_ts=1618580592.006800&cid=C01NCHR19SB

We need to add consistency between the span provider and metric provider. Currently for metrics you have to initialise your own `MeterProviderSdk` which is not necessary for spans.

Desired syntax:
> OpenTelemetrySDK.instance.meterProvider.addMetricProcessor(metricProcessor)

### PR DESCRIPTION
**Pull request by nachoBonafonte**
Add some consistency between the span provider and metric provider, so the metric provider can be configured without needing to use the constructor.
Also add support for multiple metric exporters
Fixes #167

### CODE REVIEW

**Comment by nachoBonafonte** [comment]
I know this API is highly experimental, but usability was really bad without it

**Review by Sherlouk — APPROVED** [review]
Got a super busy week ahead of me so may not be able to test it immediately in our app but the code itself looks great.

File: Sources/OpenTelemetrySdk/Metrics/Export/MetricExporter.swift [hunk 1]
```diff
@@ -19,12 +19,48 @@ public enum MetricExporterResultCode {
     case success
     case failureNotRetryable
     case failureRetryable
+
+    /// Merges the current result code with other result code
+    /// - Parameter newResultCode: the result code to merge with
+    public mutating func mergeResultCode(newResultCode: MetricExporterResultCode) {
```
Comment (SergeyKanzhelev) [inline Sources/OpenTelemetrySdk/Metrics/Export/MetricExporter.swift:25]: does it have to be public?

**Review by SergeyKanzhelev — APPROVED** [review]

File: Sources/OpenTelemetrySdk/Metrics/Export/MetricExporter.swift (hunk 1 above)
Comment (nachoBonafonte) [reply Sources/OpenTelemetrySdk/Metrics/Export/MetricExporter.swift]: No, it does not. Changed this and also the one in SpanExporterResultCode.  Thanks a lot.

**Pull request merged**